### ISSUE TITLE
Not using the current directory

### ISSUE THREAD

**Issue by ghost** [tpope/vim-fugitive#45]
First of all: Really amazing plugin! Thank you for  for that!

I've just discovered a little bug:

1) I startet vim, 
2) changed the working directory 
3) and called ":Git pull"

Git pull resists in the previos directory.

**Comment by still-dreaming-1** [comment]
I realize this isn't supported and Fugitive cares bout files, not the current working directory. I agree that fugitive should continue to be file based, I don't want that to change. But just as when you first start Vim, that is a special exception, this particular situation should also be a special exception. It would be a nice feature even if it required another special case hack.

**Comment by tpope** [comment]
There isn't a good autocommand to hook into.

**Comment by still-dreaming-1** [comment]
@tpope After doing some research, I came to the same conclusion: https://github.com/vim/vim/issues/868

**Comment by still-dreaming-1** [comment]
How would I put a hack for this in my vim config file? This is not an ideal solution, but it is the best I can come up with until someone hopefully adds an autocmd for this. I created a function to change the current working directory and I try to use that as much as possible instead of the built in way. Then from inside that function, I can have extra code that runs at the end that is a reaction to the "event" of the current working directory changing. Partly based on the way Fugitive responds to the VimEnter event, I added some code to try to get fugitive to use a different directory. But I don't totally understand it and it is not doing what I want. Here is the function.

function! ChangeDirectoryCustom(dir_path)
	let before_dir= getcwd()
	execute 'cd '.fnameescape(a:dir_path)
	let after_dir= getcwd()
	if before_dir !=# after_dir
		let current_file_name= expand('%')
		if current_file_name == ''
			call fugitive#detect(getcwd())
		endif
	endif
endfunction
